### DISCUSSION TITLE
Can we get inferred types that know the values of a union of literals?

### DISCUSSION THREAD

**Discussion by brandonpittman** [open-circle/valibot#922] [Q&A] [answered]
I've got a schema like this:

```ts
export const RecognitionOrderSchema = union([
  ...range(1, 20).map(literal),
] as const);
export type RecognitionOrder = InferOutput<typeof RecognitionOrderSchema>;
```

Things work but when I reference the `RecognitionOrder` type in Qwik it gives a type error because the type is `Literal` (which wouldn't be serializable) instead of `1 | 2 | 3 ...`.

Is there a way to infer a type that spits out the expanded union of literals instead of just `Literal`?

**Comment by brandonpittman** [comment, marked as answer]
Turns out if I manually add each literal value instead of generating them, the inferred types have real unions.

**Reply by fabian-hiller** [reply]
Yes, it seems that TS can't correctly infer the type in your first example. If you still have problems, feel free to provide a Valibot playground link.

**Reply by brandonpittman** [reply]
It’s fine. Just trying to DRY up the code. Thanks for looking.
